plugins/webp-uploads/helper.php

```diff
@@ -348,3 +354,33 @@ function webp_uploads_should_discard_additional_image_file( array $original, arr
 	}
 	return false;
 }
+
+/**
+ * Checks if a mime type is supported by the server.
+ *
+ * Includes special handling for false positives on AVIF support.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $mime_type The mime type to check.
+ *
+ * @return bool Whether the server supports a given mime type.
+ */
+function webp_uploads_mime_type_supported( string $mime_type ): bool {
+	if ( ! wp_image_editor_supports( array( 'mime_type' => $mime_type ) ) ) {
+		return false;
+	}
+
+	// In certain server environments Image editors can report a false positive for AVIF support.
+	if ( 'image/avif' === $mime_type ) {
+		$editor = _wp_image_editor_choose( 'image/avif' );
+		if ( 'WP_Image_Editor_GD' === $editor ) {
+			return function_exists( 'imageavif' );
+		}
+		if ( 'WP_Image_Editor_Imagick' === $editor ) {
+			return ! empty( Imagick::queryformats('AVIF') );
```

Comment:
Would this be better?

```suggestion
			return (bool) Imagick::queryformats('AVIF');
```

Reply:
I'm sure it must be because you are suggesting it :) Curious why though?

Do you think its clearer/easier to read? The function literally returns an empty array when AVIF is not available, so an empty check seems very appropriate and I tend to prefer empty anyway because it works well across types.

Reply:
I've just been reminded over and over that using `empty()` can be bad:

* https://localheinz.com/articles/2023/05/10/avoiding-empty-in-php/
* https://dev.to/aleksikauppila/using-isset-and-empty-hurts-your-code-aaa
* https://www.beberlei.de/post/when_to_use_empty_in_php_i_say_never

Reply:
If we know that `Imagick::queryformats('AVIF')` returns an array, and an empty array casted to `bool` is `false`, then this will work as well. Alternatively, this would explicitly be what is being checked:

```suggestion
			return count( Imagick::queryformats('AVIF') ) !== 0;
```

Reply:
> I've just been reminded over and over that using empty() can be bad:

Ok, somehow I got the opposite advice! Thanks for sharing the links, that is helpful to get the perspective. Sure, I'll go ahead and change it.

Reply:
used count approach in [`be67798` (#1176)](https://github.com/WordPress/performance/pull/1176/commits/be67798d2fa3404f66fc3596949112356b68abbc)